### PR TITLE
Add a sample Swagger UI for the CatalogService

### PR DESCRIPTION
**Pull request by beckermarc**

### CODE REVIEW

File: srv/src/main/java/my/bookshop/swagger/SwaggerSecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+
+@Configuration
+@EnableWebSecurity
+@Order(1)
+public class SwaggerSecurityConfig extends WebSecurityConfigurerAdapter {
+
+	@Override
+	protected void configure(HttpSecurity http) throws Exception {
+		http.requestMatchers().antMatchers("/swagger/**").and().authorizeRequests().anyRequest().permitAll();
```
Comment (mofterdinger) [inline srv/src/main/java/my/bookshop/swagger/SwaggerSecurityConfig.java:16]: Is this code required to enforce authentication for the swagger UI content? Or should we allow anonymous access to the swagger UI?

Comment (beckermarc) [reply srv/src/main/java/my/bookshop/swagger/SwaggerSecurityConfig.java:16]: This code enables anonymous access to the swagger UI. This is required as Spring Security (and also our Security configs) by default requires authentication for all endpoints once enabled.

Comment (mofterdinger) [reply srv/src/main/java/my/bookshop/swagger/SwaggerSecurityConfig.java:16]: ok, then it looks good to me.

**Review by mofterdinger — APPROVED** [review]

**Comment by A0094255** [comment]
Hi Marc,

In our CAP project, we have cds end points as well as spring-boot controller end points. Based on the capire [documentation ](https://cap.cloud.sap/docs/advanced/publishing-apis/openapi), we were able to generate swagger documentation for cds endpoints. How do we generate the openapi3.json file for the sprint-boot controller end point ?

**Pull request merged**